pkg/tenant/topics_test.go

```diff
@@ -1 +1,159 @@
 package tenant_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+)
+
+func (suite *tenantTestSuite) TestTopicDetail() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	topic := &db.Topic{
+		ProjectID: ulid.MustParse("01GNA91N6WMCWNG9MVSK47ZS88"),
+		ID:        ulid.MustParse("01GNA926JCTKDH3VZBTJM8MAF6"),
+		Name:      "topic-example",
+	}
+
+	defer cancel()
+
+	// Connect to mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Marshal the data with msgpack.
+	data, err := topic.MarshalValue()
+	require.NoError(err, "could not marshal the toopic")
+
+	// Unmarshal the data with msgpack.
+	other := &db.Topic{}
+	err = other.UnmarshalValue(data)
+	require.NoError(err, "could not unmarshal the topic")
+
+	// Call OnGet method and return a GetReply.
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return &pb.GetReply{
+			Value: data,
+		}, nil
+	}
+
+	// Should return an error if the topic does not exist.
+	_, err = suite.client.TopicDetail(ctx, "invalid")
+	suite.requireError(err, http.StatusBadRequest, "could not parse topic id", "expected error when topic does not exist")
+
+	// Create a topic test fixture.
+	req := &api.Topic{
+		ID:   "01GNA926JCTKDH3VZBTJM8MAF6",
+		Name: "topic-example",
+	}
+
+	rep, err := suite.client.TopicDetail(ctx, req.ID)
+	require.NoError(err, "could not retrieve topic")
+	require.Equal(req.ID, rep.ID, "expected topic ID to match")
+	require.Equal(req.Name, rep.Name, "expected topic name to match")
+
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return nil, errors.New("key not found")
+	}
+
+	_, err = suite.client.TopicDetail(ctx, "01GNA926JCTKDH3VZBTJM8MAF6")
+	suite.requireError(err, http.StatusNotFound, "could not retrieve topic", "expected error when topic ID is not found")
+}
+
+func (suite *tenantTestSuite) TestTopicUpdate() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	topic := &db.Topic{
+		ProjectID: ulid.MustParse("01GNA91N6WMCWNG9MVSK47ZS88"),
+		ID:        ulid.MustParse("01GNA926JCTKDH3VZBTJM8MAF6"),
+		Name:      "topic-example",
+	}
+
+	defer cancel()
+
+	// Connect to mock trtl database.
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Marshal the data with msgpack.
+	data, err := topic.MarshalValue()
+	require.NoError(err, "could not marshal the data")
+
+	other := &db.Topic{}
+	err = other.UnmarshalValue(data)
+	require.NoError(err, "could not unmarshal the data")
```

Comment:
would it be better to change this to `could not unmarshall the topic`?

Reply:
I'll update this as well to state `topic data.`